### DISCUSSION TITLE
Tesseract folders

### DISCUSSION THREAD

**Discussion by Perkolator** [SubtitleEdit/subtitleedit#5158] [Q&A] [answered]
I'm using the portable SubtitleEdit (currently 3.6.1) and over the years several "Tesseract" folders have been created. Which of the following folders can be safely removed?

```
Tesseract
Tesseract302
Tesseract4
Tesseract410
Tesseract411
Tesseract500.Alpha.20200328
Tesseract500.Alpha.20210506
```
I guess I should leave at least the "Tesseract302" and "Tesseract500.Alpha.20210506" folders.

The whole SubtitleEdit folder takes 1.1 GB space and all the Tesseract folders combined 647 MB space.

**Comment by niksedk** [comment, marked as answer]
Wow, that's a lot of disk space... and a lot of tesseract versions :)

Yes, keep "Tesseract302" and "Tesseract500.Alpha.20210506" folders - and delete the rest
 (as a precaution you might check if you have saved any custom files in these folders - like custom Tesseract dictionaries)


Hm, you might also want to take a backup copy of Settings.xml and then delete it... if you still have the default settings from a very old SE version, some updated default values would probably be a good thing...

**Reply by Perkolator** [reply]
> Wow, that's a lot of disk space... and a lot of tesseract versions :)

VLC and ffmpeg take a lot of space. But with shaving off the unneeded Tesseract folders I can save ~400 MB.

> as a precaution you might check if you have saved any custom files in these folders - like custom Tesseract dictionaries

Do you mean the various "*.traineddata" files? Why would I want to save these? They don't seem to hold any personal data.

> Hm, you might also want to take a backup copy of Settings.xml and then delete it... if you still have the default settings from a very old SE version, some updated default values would probably be a good thing...

Updating this file for the portable SE is a PITA. I usually forget it because the zip package doesn't have a default "Settings.xml" file. I just take the new version zip, unpack it and use file/folder compare software to check what's updated and copy/merge new stuff. Also the "Dictionaries" folder is a drag to update since few files that have been updated also contain user data in the old files (e.g. "en_names.xml").

***

Incidentally this time I remembered to compare/update the Settings.xml file. I merged few settings from the new version (though the old setting values didn't appear as something that would break something in new versions) but mostly there's just settings that I don't want to reset/change, except few that I couldn't figure out what they are for:

`AutoGuessAnsiEncoding` Should this be `True`?

`OcrBinaryImageCompareRgbThreshold` I have no idea what this is for although tried to search it from the GUI. The default value is `200` and my settings file has `270`. What does it do and should I change it?

`DCinemaFadeUpTime` and `DCinemaFadeDownTime` default is `0` and I have `5`.

`SessionKey` What's this for? Should I change it to the new default?

`ItalicFactor` I have `0.2` and default is `0.200000002980232`.

***

P.S. Feature requests:
- Remember "Multiple replace" window size and the "splitter" position.
- Remember "Start time" and "End time" column widths in main and OCR window. (The columns in the main window seem to keep the widths most of the time but randomly (I haven't found a reason why) the column widths are reset to default.)

Thanks

**Reply by niksedk** [reply]
OK, sounds like you are in full control if changes :)
`AutoGuessAnsiEncoding` should probably be `True` as it works better now (opening ANSI files, and luckily ppl are more and more often just saving in UTF-8 :)
The other settings are probably not important... `SessionKey` is for online collaboration between multiple people (not used much I think).

I could probably improve the dictionary stuff... but it's not high on my long long todo list.

Please put each feature request in it's own `Issue`, thx.

**Reply by Perkolator** [reply]
> * Remember "Multiple replace" window size and the "splitter" position.

There actually was already [an issue for this](https://github.com/SubtitleEdit/subtitleedit/issues/2377). 4 years old, and I had even commented on it over a year ago.

> * Remember "Start time" and "End time" column widths in main and OCR window.

Found out the reason why main window columns changes sometimes. https://github.com/SubtitleEdit/subtitleedit/issues/5164
